File: week_2_arrays/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

//index = 0.0588 * L - 0.296 * S - 15.8

int main(void)
{
    float L = 0.0;
    int W = 0.0;
    float S = 0.0;

    //prompt user for input
    string userInput = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text

    for(int i = 0, len = strlen(userInput); i < len; i++)
    {
        if(isalpha(userInput[i])){
            L++;
        }
        if(userInput[i] == ' '){
            W++;
        }
        if(userInput[i] == '.' || userInput[i] == '!' || userInput[i] == '?'){
            S++;
        }
    }
    W++;
    L = (L / W) * 100;
    S = (S / W) * 100;
    float grade = 0.0588 * L - 0.296 * S - 15.8;
    int finalGrade = round(grade);

    if(finalGrade <1){
        printf("Before Grade 1\n");
    }
    else if(finalGrade >= 16){
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", finalGrade);
    }
}

File: week_6_python/sentimental-readability/readability.py
userInput = input("Text: ")
L = 0.0
W = 0.0
S = 0.0

for i in range(len(userInput)):
    if (userInput[i].isalpha()):
        L = L + 1
    if (userInput[i] == " "):
        W = W + 1
    if (userInput[i] in [".", "!", "?"]):
        S = S + 1


W = W + 1

L = (L / W) * 100
S = (S / W) * 100
grade = 0.0588 * L - 0.296 * S - 15.8
finalGrade = int(round(grade))

if (finalGrade < 1):
    print("Before Grade 1")
elif (finalGrade >= 16):
    print("Grade 16+")
else:
    print(f"Grade {finalGrade}")
